src/libs/actions/http-request.ts

```diff
@@ -0,0 +1,160 @@
+  saveHttpRequestActionConfigs()
+  updateCockpitActions()
+}
+
+export const updateHttpRequestActionConfig = (id: string, updatedAction: HttpRequestActionConfig): void => {
+  registeredHttpRequestActionConfigs[id] = updatedAction
+  saveHttpRequestActionConfigs()
+  updateCockpitActions()
+}
+
+export const updateCockpitActions = (): void => {
+  Object.entries(availableCockpitActions).forEach(([id]) => {
+    if (id.includes(httpRequestActionIdPrefix)) {
+      deleteAction(id as CockpitActionsFunction)
+    }
+  })
+
+  const httpResquestActions = getAllHttpRequestActionConfigs()
+  for (const [id, action] of Object.entries(httpResquestActions)) {
+    try {
+      const cockpitAction = new CockpitAction(id as CockpitActionsFunction, action.name)
+      registerNewAction(cockpitAction)
+      registerActionCallback(cockpitAction, getHttpRequestActionCallback(id))
+    } catch (error) {
+      console.error(`Error registering action ${id}: ${error}`)
+    }
+  }
+}
+
+export const loadHttpRequestActionConfigs = (): void => {
+  const savedActions = localStorage.getItem('cockpit-http-request-actions')
+  if (savedActions) {
+    registeredHttpRequestActionConfigs = JSON.parse(savedActions)
+  }
+}
+
+export const saveHttpRequestActionConfigs = (): void => {
+  localStorage.setItem('cockpit-http-request-actions', JSON.stringify(registeredHttpRequestActionConfigs))
+}
+
+export type HttpRequestActionCallback = () => void
+
+export const getHttpRequestActionCallback = (id: string): HttpRequestActionCallback => {
+  const action = getHttpRequestActionConfig(id)
+  if (!action) {
+    throw new Error(`Action with id ${id} not found.`)
+  }
+
+  let parsedBody = action.body
+  const parsedUrlParams = action.urlParams
+
+  const cockpitInputsInBody = action.body.match(/{{\s*([^{}\s]+)\s*}}/g)
+  if (cockpitInputsInBody) {
+    for (const input of cockpitInputsInBody) {
+      const parsedInput = input.replace('{{', '').replace('}}', '').trim()
+      const inputData = getCockpitActionParameterData(parsedInput)
+      if (inputData) {
+        parsedBody = parsedBody.replace(input, inputData.toString())
+      }
+    }
+  }
+
+  const cockpitInputsInUrlParams = Object.entries(action.urlParams).filter(
+    ([, value]) => typeof value === 'string' && value.startsWith('{{') && value.endsWith('}}')
+  )
+  if (cockpitInputsInUrlParams) {
+    for (const [key, value] of cockpitInputsInUrlParams) {
+      const parsedInput = value.replace('{{', '').replace('}}', '').trim()
+      const inputData = getCockpitActionParameterData(parsedInput)
+      if (inputData) {
+        parsedUrlParams[key] = inputData.toString()
+      }
+    }
+  }
+
+  const url = new URL(action.url)
+
+  url.search = new URLSearchParams(parsedUrlParams).toString()
+
+  return () => {
+    fetch(url, {
+      method: action.method,
+      headers: action.headers,
+      body: action.method === HttpRequestMethod.GET ? undefined : parsedBody,
+    })
+  }
+}
```

Comment:
The parameters for the body and url variables are registered when the callback is created, but if they change by the time its executed, the new values won't be updated.

Maybe moving them inside the return, like:
```
export const getHttpRequestActionCallback = (id: string): HttpRequestActionCallback => {
  const action = getHttpRequestActionConfig(id)
  if (!action) {
    throw new Error(`Action with id ${id} not found.`)
  }

  return () => {
    let parsedBody = action.body
    const parsedUrlParams = { ...action.urlParams }

 [...]

    fetch(url, {
      method: action.method,
      headers: action.headers,
      body: action.method === HttpRequestMethod.GET ? undefined : parsedBody,
    })
  }
}
```